src/Core/Checkout/Customer/CustomerValueResolver.php

```diff
@@ -23,14 +24,24 @@ public function resolve(Request $request, ArgumentMetadata $argument): \Generato
         if ($loginRequired !== true) {
             $route = $request->attributes->get('_route');
 
-            throw new \RuntimeException('Missing @LoginRequired annotation for route: ' . $route);
+            // @deprecated tag:v6.8.0 - remove this if block
+            if (!Feature::isActive('v6.8.0.0')) {
+                // @phpstan-ignore-next-line
+                throw new \RuntimeException('Missing @LoginRequired annotation for route: ' . $route);
```

Comment:
it is not necessary to put throwing of a RuntimeException around a feature flag. the domain exceptions are extending in the end from RuntimeException, so whoever catch that, can still do so, as it won't break